src/sentry/static/sentry/app/components/pill.jsx

```diff
@@ -29,19 +29,84 @@ class Pill extends React.Component {
   };
 
   render() {
-    const {name, children, className, ...props} = this.props;
+    const {name, children, ...props} = this.props;
     const [extraClass, renderedValue] = this.renderValue();
-
     return (
-      <li className={classNames(className, extraClass)} {...props}>
-        <span className="key">{name}</span>
-        <span className="value">
+      <StyledPill {...props}>
+        <PillName>{name}</PillName>
+        <PillValue type={extraClass}>
           {renderedValue}
           {children}
-        </span>
-      </li>
+        </PillValue>
+      </StyledPill>
     );
   }
 }
 
+const StyledPill = styled('li')`
+  white-space: nowrap;
+  margin: 0 10px 10px 0;
+  display: flex;
+  border: 1px solid ${p => p.theme.gray1};
+  border-radius: 3px;
+  box-shadow: ${p => p.theme.dropShadowLightest};
+  line-height: 1.2;
+  max-width: 100%;
+  &:last-child {
+    margin-right: 0;
+  }
+`;
+
+const PillName = styled('span')`
+  padding: ${space(0.5)} ${space(1)};
+  min-width: 0;
+  white-space: nowrap;
+`;
+
+const PillValue = styled(PillName)`
+  background: ${p => p.theme.whiteDark};
+  border-left: 1px solid ${p => p.theme.gray1};
+  border-radius: 0 3px 3px 0;
+  font-family: ${p => p.theme.text.familyMono};
+  max-width: 100%;
+  > a {
+    max-width: 100%;
+    text-overflow: ellipsis;
```

Comment:
can we use `app/styles/overflowEllipsis` here?

Reply:
There is an issue with the pills that have an .extrenal-icon inline _(open & info)_ concerning the width and I would need to modify the styles in the overflowEllipsis.